chains/ethereum/connection_test.go

```diff
@@ -125,3 +129,85 @@ func createTestAuth(t *testing.T, conn *Connection) *bind.TransactOpts {
 
 	return auth
 }
+
+func panicOnTimeout(d time.Duration) {
+	<-time.After(d)
+	panic("Test timed out")
+}
+
+// TestContractCode is used to make sure the contracts are deployed correctly.
+// This is probably the least intrusive way to check if the contracts exists
+func TestContractCode(t *testing.T) {
+
+	// We set a custom timeout for this test
+	// This is because of a ganache bug
+	// To reproduce the bug, attempt to create a contract Object (eg Emitter) with a valid Address of a different contract type (eg RecieverAddress)
+	// It will cause the test to hang forever, this code is to prevent the hanging in that case.
+	go panicOnTimeout(TestTimeout)
```

Comment:
The contract method calls below block, because the address is incorrect. I feel like this is really a symptom of the problem we are trying to solve. 

I've opened another issue (#184) about how we can introduce the deployed bytecode to the system. I think it would be best to assume these tests pass and remove this, with the understanding that once we can compare the bytecode this case should never happen.

Reply:
Once we have the deployed bytecode, this test become a lot cleaner and it would be a lot easier to implement a function to check contract correctness on runtime. I will make this PR as blocked on those.